terraform/deployments/govuk-eks/helm_auth.tf

```diff
@@ -0,0 +1,11 @@
+provider "helm" {
```

Comment:
Something to be aware of here - providers defined with attribute values from other provider's resources are [not predictable](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#stacking-with-managed-kubernetes-cluster-resources) and can lead to hard to reproduce issues, so the recommendation is to handle EKS and Helm/Kubernetes resources in completely separate terraform states and apply phases.